main.cc

```diff
@@ -601,6 +610,12 @@ struct add_block_entity_tool : tool
         if (!rc->hit || rc->inside)
             return false;
 
+        /* don't allow placements that would cause the player to end up inside the ent and get stuck */
```

Comment:
glad we have captured this case, the other day playing I was trying to make it happen and (by luck it seems) wasn't able to trap myself.